src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,310 @@
+import de.hpi.bpt.argos.util.RestEndpointUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EntityEndpointImpl implements EntityEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final Gson serializer = new Gson();
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.get(EntityEndpoint.getEntityBaseUri(), this::getEntity);
+        sparkService.get(EntityEndpoint.getChildEntitiesBaseUri(), this::getChildEntities);
+        sparkService.get(EntityEndpoint.getEventTypesOfEntityBaseUri(), this::getEventTypesOfEntity);
+        sparkService.get(EntityEndpoint.getEventsOfEntityBaseUri(), this::getEventsOfEntity);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEntity(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        Entity eventType = PersistenceAdapterImpl.getInstance().getEntity(entityId);
+        List<Attribute> attributes = PersistenceAdapterImpl.getInstance().getAttributes(entityId);
+        JsonObject jsonEventType = getEntityJson(eventType, attributes);
+
+        response.body(serializer.toJson(jsonEventType));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getChildEntities(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        long entityTypeId = getEntityTypeId(request);
+        List<Entity> childEntities = PersistenceAdapterImpl.getInstance().getEntities(entityId, entityTypeId);
+        List<Attribute> attributesToInclude = getAttributesToInclude(request);
+        JsonObject jsonEntities = getEntitiesJson(childEntities, attributesToInclude);
+
+        response.body(serializer.toJson(jsonEntities));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypesOfEntity(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityId = getEntityId(request);
+        Entity entity = PersistenceAdapterImpl.getInstance().getEntity(entityId);

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "entity" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "entity". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1854)